src/commands/protocols/get/sender/local-get-command.js

```diff
@@ -80,14 +80,37 @@ class LocalGetCommand extends Command {
         // }
 
         // else {
+        this.operationIdService.emitChangeEvent(
+            OPERATION_ID_STATUS.GET.GET_LOCAL_GET_ASSERTION_START,
+            operationId,
+            blockchain,
+        );
+        const assertion = await this.tripleStoreService.getAssertion(ual);
+        // const promises = [this.tripleStoreService.getAssertion(ual)];
+        this.operationIdService.emitChangeEvent(
+            OPERATION_ID_STATUS.GET.GET_LOCAL_GET_ASSERTION_END,
+            operationId,
+            blockchain,
+        );
 
-        const promises = [this.tripleStoreService.getAssertion(ual)];
-
-        if (includeMetadata) {
-            promises.push(this.tripleStoreService.getKnowledgeAssetMetadata(ual));
-        }
+        this.operationIdService.emitChangeEvent(
+            OPERATION_ID_STATUS.GET.GET_LOCAL_GET_KA_METADATA_START,
+            operationId,
+            blockchain,
+        );
+        const knowledgeAssetMetadata = includeMetadata
+            ? await this.tripleStoreService.getKnowledgeAssetMetadata(ual)
+            : undefined;
+        this.operationIdService.emitChangeEvent(
+            OPERATION_ID_STATUS.GET.GET_LOCAL_GET_KA_METADATA_END,
+            operationId,
+            blockchain,
+        );
+        // if (includeMetadata) {
```

Comment:
This needs to be done in promises, you cant because of measuring speed to make things run slower